userspace/libscap/scap_procs.c

```diff
@@ -1002,6 +1002,16 @@ static int32_t _scap_proc_scan_proc_dir_impl(scap_t* handle, char* procdirname,
 		if(res != SCAP_SUCCESS)
 		{
 			snprintf(error, SCAP_LASTERR_SIZE, "cannot add procs tid = %"PRIu64", parenttid = %"PRIi32", dirname = %s, error=%s", tid, parenttid, procdirname, add_error);
+			//
+			// Bug 1897: Agent fails to initialize sinsp_worker: error creating the process list
+			//           When parameter suppress is true, report in the log, but do not process 
+			//           this file any further, because it is no longer in the process list.  
+			//           This is not considered a failure.
+			//
+			if(suppress)
+			{
+				res = SCAP_SUCCESS;
+			}
 			break;
```

Comment:
I very much like the comment, and I think this is a big improvement. The break statement should be in an else block though, because we only want to break if we encountered an error.

Reply:
An else block will be placed around the break, so we remain in the while loop.